pkg/provider/aws/aws.go

```diff
@@ -83,7 +83,7 @@ type Client struct {
 
 func New(log *logger.FunLogger, env v1alpha1.Environment, cacheFile string) (*Client, error) {
 	// Create an AWS session and configure the EC2 client
-	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(env.Spec.Region))
+	cfg, err := config.LoadDefaultConfig(context.TODO())
```

Comment:
Does this mechanism not work to set these options?

Reply:
with that set, it default to config file first, then env vars.

Reply:
I don't quite understand. There are three ways to set the AWS region here.

1. The config file as defined in this project.
2. The AWS config file.
3. The `AWS_REGION` environment variable.

In what order to we want to apply them? Why is the region special compared to other environment variables?

Reply:
before this PR, this project was only taking into account the region in the config file. 
This PR fixes that by moving to 
- ENV AWS_REGION 
- Config file
aws config file is not taken into account, since the idea is to be used by a github action or manual testing.

Reply:
Thanks for the suggestion, change added